### PR TITLE
Add Kernel32.GetProcessTimes

### PR DESCRIPTION
**Pull request by adrianrus**

### CODE REVIEW

File: src/Kernel32.Tests/storebanned/Kernel32Facts.cs
Comment (AArnott) [inline src/Kernel32.Tests/storebanned/Kernel32Facts.cs]: Aren't you inviting a race condition here between the two queries for this value that can lead to an unstable test?
Can we reduce the precision of the equality check perhaps so that anything within a minute or so will pass?

Comment (adrianrus) [reply src/Kernel32.Tests/storebanned/Kernel32Facts.cs]: yes, there was a race condition + some conversion issues

part of the conversion issue is that kernel32.FILETIME has the dwLowDateTime and dwHighDateTime reversed?!  

check https://msdn.microsoft.com/en-us/library/windows/desktop/ms724284(v=vs.85).aspx or 
https://referencesource.microsoft.com/#mscorlib/system/runtime/interopservices/ComTypes/imoniker.cs,1a6d8e76ae2c94a1
vs 
https://github.com/AArnott/pinvoke/blob/bc4c6830f1918f5af28e76e075dd0750f27d3c61/src/Kernel32/Kernel32%2BFILETIME.cs

**Comment by adrianrus** [comment]
please don't approve the pull request until we reach a conclusion with the order of fields in FILETIME structure

File: src/Kernel32.Tests/storebanned/Kernel32Facts.cs
Comment (adrianrus) [reply src/Kernel32.Tests/storebanned/Kernel32Facts.cs]: using long (like here => https://referencesource.microsoft.com/#System/compmod/microsoft/win32/NativeMethods.cs,924eca64608eba79) instead of Kernel32.FILETIME  structure fixes the reversed fields issue, but that seems a error which needs to be addressed.

Comment (AArnott) [reply src/Kernel32.Tests/storebanned/Kernel32Facts.cs]: #417 will fix the `FILETIME` issue. Will you want to keep using `long` or switch to `FILETIME`?

Comment (AArnott) [reply src/Kernel32.Tests/storebanned/Kernel32Facts.cs]: Looking over your latest iteration, I think casting as `FILETIME` would make it a more usable API. Can you switch it back after merging/rebasing on #417 (either now, or once it's merged in)?

Comment (adrianrus) [reply src/Kernel32.Tests/storebanned/Kernel32Facts.cs]: OK, but the cast is not straightforward ... check the UT

Comment (AArnott) [reply src/Kernel32.Tests/storebanned/Kernel32Facts.cs]: Is the converstion from `FILETIME` to `long` common? Is it any easier when using the .NET-defined ComTypes one?

What if we built the casting into our custom `FILETIME` struct so that it had a `(long)` cast operator?

Comment (adrianrus) [reply src/Kernel32.Tests/storebanned/Kernel32Facts.cs]: Seems a good idea to have the cast operator, with that you can get quickly to a TimeSpan or DateTime

Comment (AArnott) [reply src/Kernel32.Tests/storebanned/Kernel32Facts.cs]: Do you want to add that to your PR?

Comment (adrianrus) [reply src/Kernel32.Tests/storebanned/Kernel32Facts.cs]: done.

**Review by AArnott — APPROVED** [review]
Awesome! Thanks.

**Pull request merged**
